### ISSUE TITLE
 Windows - Nested Source Folder Causes Exception: TypeError: (intermediate value) is not iterable - when there are Tabs in the cabal file

### ISSUE THREAD

**Issue by theGhostJW** [atom-haskell/ide-haskell-repl#68]
[Enter steps to reproduce:]

1. Cabal file
   
        hs-source-dirs:      src
				,  src\MTL
				,  src\FreeMonad
          main-is:             Main.hs
          other-modules:   MTLPlusLens
                               , ProgrammingWithMonadsRWH
                                , HidingImp1RWH
                               , HandleIONewTypeRWH
                              , FreeMonad
        default-language:    Haskell2010

2. Stack Build from Cmd line works
3. Open Repl in of Atom (any source file)
=> gets the following error

**Atom**: 1.28.2 x64
**Electron**: 2.0.5
**OS**: Microsoft Windows 10 Pro
**Thrown From**: [ide-haskell-repl](https://github.com/atom-haskell/ide-haskell-repl) package 0.8.1

### Stack Trace

TypeError: (intermediate value) is not iterable

```
At TypeError: (intermediate value) is not iterable

TypeError: (intermediate value) is not iterable
    at Function.componentFromURI (/packages/ide-haskell-repl/lib/ide-haskell-repl-base.js:52:22)
    at <anonymous>
```

### Non-Core Packages

```
atom-beautify 0.32.5 
atom-haskell 0.4.0 
atom-ide-ui 0.13.0 
atom-js-console 1.2.2 
atom-save-all 0.4.1 
atom-ternjs 0.18.3 
auto-detect-indentation 1.3.0 
autocomplete-haskell 1.0.1 
autocomplete-modules 2.1.0 
autoprefixer 3.7.1 
ava 0.11.0 
browser-plus 0.0.98 
browser-plus-zoom 0.0.2 
busy-signal 1.4.3 
clipboard-plus 0.5.1 
color-picker 2.3.0 
copy-find-results 0.1.0 
docblockr 0.13.7 
drag-drop-text 0.1.4 
editorconfig 2.2.2 
emmet 2.4.3 
file-icons 2.1.22 
flow-ide 1.11.0 
git-time-machine 2.1.0 
haskell-ghc-mod 2.2.4 
haskell-pointfree 0.2.0 
haskell-scry 1.0.8 
hasklig 0.4.0 
hey-pane 1.1.0 
highlight-selected 0.14.0 
hyperclick 0.0.0 
ide-flowtype 0.23.1 
ide-flowtype-coverage 1.9.2 
ide-haskell 2.3.1 
ide-haskell-cabal 2.2.0 
ide-haskell-hasktags 0.0.15 
ide-haskell-hie 0.10.1 
ide-haskell-hoogle 0.1.2 
ide-haskell-repl 0.8.1 
ide-purescript 0.24.0 
intentions 1.1.5 
language-ansi 1.1.1 
language-babel 2.85.0 
language-graphql 0.10.1 
language-haskell 1.17.8 
language-haskell-scoped 0.1.1 
language-ini 1.20.0 
language-lua 0.9.11 
language-ocaml 1.9.5 
language-purescript 0.8.3 
language-rust 0.4.12 
language-swift 0.5.0 
language-thrift 1.0.2 
linter 2.2.0 
linter-eslint 8.4.1 
linter-flow 5.6.1 
linter-ui-default 1.7.1 
MagicPython 1.0.12 
markdown-scroll-sync 2.1.2 
merge-conflicts 1.4.5 
minimap 4.29.8 
minimap-autohide 0.10.1 
minimap-bookmarks 0.4.2 
minimap-codeglance 0.4.7 
minimap-highlight-selected 4.6.1 
minimap-selection 4.5.0 
minimap-split-diff 0.3.7 
navigate 0.1.20 
navigation-history 0.1.8 
npm 2.0.0 
nuclide 0.321.0 
nuclide-format-js 0.0.46 
nuclide-test-runner-mocha 0.1.1 
open-recent 5.0.0 
path-copy 0.14.0 
pigments 0.40.2 
platformio-ide-terminal 2.8.3 
print-atom 0.1.1 
language-Quickstep 1.4.0 
script 3.18.1 
set-syntax 0.4.0 
sort-lines 0.18.0 
split-diff 1.5.2 
Sublime-Style-Column-Selection 1.7.4 
tablr 1.8.3 
teletype 0.13.3 
todo-show 2.3.0 
tomatimer 1.2.3 
tool-bar 1.1.7 
vertical-tabs 1.0.7 
zentabs 0.8.9 
```

**Work Around**
Remove sub dirs from cabal file and move source files to src

    executable PyrethrumPre
      hs-source-dirs:      src
      main-is:             Main.hs

**Comment by lierdakil** [comment]
Thanks for reporting, I'll try looking into it.

In the meantime... shouldn't you be using a forward slash in paths? Since cabalfiles are supposedly cross-platform and all.

**Comment by theGhostJW** [comment]
Hmm now with one file in one sub-dir its working back or forward slash. The backslash above was an artifact of me trying to get the RPL to work I had forward slashes before.

I need to do some more playing around to reproduce though looks like there is more to it than I thought. Will do that now.

**Comment by theGhostJW** [comment]
OK its a tabs Vs spaces issue in the cabal file.
So if you have a line like:
```
\t\t\t\t\t,  src\MTL 
```
=> Stack will compile with a warning 
```
Warning: Cabal file warning in C:\PyrethrumPre\PyrethrumPre.cabal@ 17:12: Tabs used as indentation
```
but the REPL will produce the above error.

**Comment by lierdakil** [comment]
Huh. Well, according to [cabal user guide](https://www.haskell.org/cabal/users-guide/developing-packages.html#package-descriptions):
> Tabs are not allowed as indentation characters due to a missing standard interpretation of tab width.

So there's that.

**Comment by lierdakil** [comment]
The error message is cryptic though, so this case needs to be handled. Thanks for investigating.

**Comment by theGhostJW** [comment]
NP - I think (Stack / Cabal) should have thrown an error if Tabs are not allowed. But things being as they are a descriptive error from REPL would be great solution.

**Comment by lierdakil** [comment]
Should be fixed in v0.8.2. Sorry for the delay.
